libs/ledger-live-common/src/market/v2/useMarketDataProvider.ts

```diff
@@ -0,0 +1,128 @@
+import { useQueries, useQuery } from "@tanstack/react-query";
+import {
+  fetchCurrency,
+  fetchCurrencyChartData,
+  fetchCurrencyData,
+  fetchList,
+  getSupportedCoinsList,
+  supportedCounterCurrencies,
+} from "../api/api";
+import {
+  MarketCurrencyRequestParams,
+  MarketListRequestParams,
+  MarketListRequestResult,
+} from "../types";
+import { QUERY_KEY } from "./queryKeys";
+import { listSupportedCurrencies } from "../../currencies";
+import { CryptoCurrency } from "@ledgerhq/types-cryptoassets";
+import { useMemo } from "react";
+
+// 1min
+const REFETCH_TIME = 1 * 60000;
+
+export function useMarketDataProvider() {
+  const supportedCurrenciesInLIve = listSupportedCurrencies();
+
+  const liveCompatibleIds: string[] = supportedCurrenciesInLIve
+    .map(({ id }: CryptoCurrency) => id)
+    .filter(Boolean);
+  const { data: supportedCounterCurrencies } = useSupportedCounterCurrencies();
+  const { data: supportedCurrencies } = useSupportedCurrencies();
+
+  const liveCoinsList = useMemo(
+    () =>
+      supportedCurrencies?.filter(({ id }) => liveCompatibleIds.includes(id)).map(({ id }) => id),
+    [liveCompatibleIds, supportedCurrencies],
+  );
+
+  return {
+    supportedCounterCurrencies,
+    supportedCurrencies,
+    liveCoinsList,
+  };
+}
+
+export function useCurrencyChartData({ id, counterCurrency, ranges }: MarketCurrencyRequestParams) {
+  const results = useQueries({
+    queries: (ranges || []).map(range => ({
+      queryKey: [QUERY_KEY.CurrencyChartData, id, counterCurrency, range],
+      queryFn: () => fetchCurrencyChartData({ counterCurrency, range, id }),
+      refetchInterval: REFETCH_TIME,
+      staleTime: REFETCH_TIME,
+    })),
+  });
+
+  return results;
+}
+
+export function useCurrencyData({ id, counterCurrency, ranges }: MarketCurrencyRequestParams) {
+  const results = useQueries({
+    queries: (ranges || []).map(range => ({
+      queryKey: [QUERY_KEY.CurrencyData, id, counterCurrency, range],
+      queryFn: () => fetchCurrencyData({ counterCurrency, range, id }),
+      refetchInterval: REFETCH_TIME,
+      staleTime: REFETCH_TIME,
+    })),
+  });
+
+  const resultCurrency = useQuery({
+    queryKey: [QUERY_KEY.CurrencyDataRaw, id],
+    queryFn: () => fetchCurrency({ id }),
+    refetchInterval: REFETCH_TIME,
+    staleTime: REFETCH_TIME,
+  });
+
+  return { currencyDataByRanges: results, currencyInfo: resultCurrency };
+}
+
+export function useSupportedCounterCurrencies() {
+  const { isPending, error, data, isFetching } = useQuery({
+    queryKey: [QUERY_KEY.SupportedCounterCurrencies],
+    queryFn: () => supportedCounterCurrencies(),
+  });
+
+  return {
+    isPending,
+    error,
+    data,
+    isFetching,
+  };
+}
```

Comment:
why are we abstracting a subset and returning this ... think consistency is key, and forming patterns, could be simplified to:
```suggestion
export function useSupportedCounterCurrencies() => useQuery({
    queryKey: [QUERY_KEY.SupportedCounterCurrencies],
    queryFn: () => supportedCounterCurrencies(),
  });
```